### PR TITLE
Issue #7765: XpathRegression tests added for UnnecessaryParentheses

### LINKED ISSUE #7765: Update AbstractChecks to log DetailAST - UnnecessaryParentheses
Child of https://github.com/checkstyle/checkstyle/issues/5777

This issue specifically focuses on UnnecessaryParentheses.

### PR DESCRIPTION
**Pull request by shashwat70**
Resolves #7765

### CODE REVIEW

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.UnnecessaryParenthesesCheck;
+
+public class XpathRegressionUnnecessaryParenthesesTest extends AbstractXpathTestSupport {
```
Comment (rdiachenko) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: Could you add more tests please. From #5777:
> 2: Must have a UT for each log in the check. If the check has 3 separate log calls, then it must have 3 separate UTs.

**Comment by rdiachenko** [comment]
@shashwat70 ping

**Comment by shashwat70** [comment]
> @shashwat70 ping

Got a bit busy with stuff. Done. @rdiachenko

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java (hunk 1 above)
Comment (rdiachenko) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: @shashwat70 
Still not done. There are 8 logs in the check but your test covers only 7. Looks like you missed this one:
```
if (assignDepth >= 1) {
    log(ast, MSG_ASSIGN);
}
```

**Comment by rdiachenko** [comment]
@shashwat70 please see a new comment above, the review note is still not fixed. Could you fix travis and IDEA inspections failures as well?

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java (hunk 1 above)
Comment (shashwat70) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: Could you please help me figure out what kind of errors are being logged by this log call? I figured something out but ended up finding a bug instead. #8239 
@rdiachenko

Comment (shashwat70) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: @rdiachenko ping

Comment (rdiachenko) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: @shashwat70 , sorry for a delayed response.
Please take a look at this test - https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheckTest.java . It should cover that branch. Put a break point and try to debug yourself. If you still have problems please let me know.

Comment (shashwat70) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionUnnecessaryParenthesesTest.java:31]: Done

**Comment by shashwat70** [comment]
> @shashwat70 please see a new comment above, the review note is still not fixed. Could you fix travis and IDEA inspections failures as well?

Done @rdiachenko

**Comment by shashwat70** [comment]
> > > @shashwat70 please see a new comment above, the review note is still not fixed. Could you fix travis and IDEA inspections failures as well?
> > 
> > 
> > Done @rdiachenko
> 
> @shashwat70 IDEA Inspections are still failing. Please fix.

Done @rdiachenko

**Pull request merged**